src/components/ApiAction/QueryCard.vue

```diff
@@ -0,0 +1,341 @@
+        <b-row align-v="stretch" class="multipaneRow px-3">
+          <Multipane
+            class="QueryLayout-vertical Query-Custom-resizer-vertical"
+            layout="vertical"
+          >
+            <div
+              class="QueryLayout-sidebarWrapper-vertical"
+              data-cy="QueryLayout-sidebarWrapper"
+            >
+              <b-card no-body class="h-100">
+                <json-editor
+                  :id="`queryEditorWrapper-${tabIdx}`"
+                  :ref="`queryEditorWrapper-${tabIdx}`"
+                  class="m-2 h-100"
+                  :data-cy="`api-actions-query-JSONEditor-${tabIdx}`"
+                  :content="jsonQuery"
+                  @change="queryBodyChange"
+                />
+              </b-card>
+            </div>
+            <MultipaneResizer data-cy="sidebarResizer" />
+            <div class="QueryLayout-contentWrapper-vertical">
+              <ResponseCard :tabIdx="tabIdx" :response="response" />
+            </div>
+          </Multipane>
+        </b-row>
+      </b-container>
+    </b-card-text>
+  </b-card-body>
+</template>
+
+<script>
+import jsonEditor from '@/components/Common/JsonEditor'
+import ResponseCard from '@/components/ApiAction/ResponseCard'
+import { Multipane, MultipaneResizer } from 'vue-multipane'
+import _ from 'lodash'
+
+export default {
+  name: 'QueryCard',
+  props: {
+    query: {},
+    tabIdx: {},
+    controllers: {},
+    actions: {},
+    api: {},
+    openapi: {},
+    response: {}
+  },
+  components: {
+    jsonEditor,
+    ResponseCard,
+    Multipane,
+    MultipaneResizer
+  },
+  data() {
+    return {
+      isFullScreen: false,
+      jsonQuery: '{}',
+      editedQuery: {
+        controller: null,
+        action: null,
+        body: {}
+      }
+    }
+  },
+  mounted() {
+    this.editedQuery = _.clone(this.query)
+    this.jsonQuery = JSON.stringify(this.editedQuery, null, 2)
+  },
+  methods: {
+    toggleFullscreen() {
+      this.isFullScreen = !this.isFullScreen
+    },
+    saveQuery() {
+      this.$emit('saveQuery', this.tabIdx)
+    },
+    performQuery() {
+      this.$emit('performQuery', this.tabIdx)
+    },
+    loadQueryParams() {
+      const query = _.clone(this.editedQuery)
+      if (
+        !this.api ||
+        !this.api[query.controller] ||
+        !this.api[query.controller][query.action]
+      ) {
+        let obj = {
+          controller: query.controller,
+          action: query.action,
+          body: query.body
+        }
+        this.jsonQuery = JSON.stringify(obj, null, 2)
+
+        this.$refs[`queryEditorWrapper-${this.tabIdx}`].setContent(
+          this.jsonQuery
+        )
+        return
+      }
+      const api = this.api[query.controller][query.action]
+      let path = api.http[0].url
+      let verb = api.http[0].verb.toLowerCase()
+      let tmpPath = path
+      let idx = tmpPath.search(/\/:/)
+      while (idx !== -1) {
+        tmpPath = tmpPath.replace('/:', '/{')
```

Comment:
there is no other simple way to extract routes params? this part is really not readable/maintainable